src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -110,9 +111,21 @@ public MaxwellKafkaProducer(MaxwellContext context, Properties kafkaProperties,
 		this.inflightMessages = new InflightMessageList();
 	}
 
+	private Integer getNumPartitions(String topic) {
+		try {
+			return this.kafka.partitionsFor(topic).size(); //returns 1 for new topics
+		} catch(TimeoutException e) {
+			LOGGER.error("Topic '" + topic + "' name does not exist. Exception: " + e.getLocalizedMessage());
+			throw e;
```

Comment:
wondering how thoroughly we've tested this; does the 0.9 producer do anything smarter when given a non-existent topic?  Or are we really stuck guessing at timeouts?